### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by joel-kalema**
- [ ] Add a column species of type string to your animals table. Modify your `schema.sql` file.
- [ ] Insert the data.
- [ ] Remember to add these insertions to your data.sql file.
- [ ] Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then 
       roll back the change and verify that species columns went back to the state before transaction.
- [ ] Inside a transaction:
       - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
       - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
       - Commit the transaction.
       - Verify that change was made and persists after commit.
- [ ] Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- [ ] After the roll back verify if all records in the animals table still exist. After that you can start breath as usual ;)
- [ ] Inside a transaction:
       - Delete all animals born after Jan 1st, 2022.
       - Create a savepoint for the transaction.
       - Update all animals' weight to be their weight multiplied by -1.
       - Rollback to the savepoint
       - Update all animals' weights that are negative to be their weight multiplied by -1.
       - Commit transaction
- [ ] Write queries to answer the following questions:
       - How many animals are there?
       - How many animals have never tried to escape?
       - What is the average weight of animals?
       - Who escapes the most, neutered or not neutered animals?
       - What is the minimum and maximum weight of each type of animal?
       - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- [ ] Remember to add these queries to your `queries.sql` file.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -17,4 +15,50 @@ select * from animals where neutered =true;
...
 select * from animals where name!='Gabumon' ;
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-select * from animals where weight_kg>=10.4 AND weight_kg<=17.3;
\ No newline at end of file
+select * from animals where weight_kg>=10.4 AND weight_kg<=17.3;
+
+-- starting a transaction and update animal table by setting the species column to unspecified and rollback.
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (brytebee) [inline queries.sql:23]: - [x] **Required:** You are doing awesome. However, you must explicitly verify that the change was implemented before and after the rollback.

_**Hint**_ Consider `SELECT species FROM animals;`

File: queries.sql [hunk 2]
```diff
@@ -17,4 +15,50 @@ select * from animals where neutered =true;
...
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals SET species = 'digimon' WHERE species = '';
+-- Commit the transaction.
+COMMIT;
```
Comment (brytebee) [inline queries.sql:33]: - [x] **Required:** You are doing awesome. However, you must explicitly verify that the change was implemented before and after the `commit`.

_**Hint**_ Consider `SELECT species FROM animals;`

File: queries.sql [hunk 3]
```diff
@@ -17,4 +15,50 @@ select * from animals where neutered =true;
...
+-- starting a transaction and update animal table by setting the species column to unspecified and rollback.
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals SET species = 'digimon' WHERE species = '';
```
Comment (brytebee) [inline queries.sql]: - [x] **Required:** At this point the column `species` eitiher has the value `digimon` or `NULL`. Using an empty string would work. Kindly update to `NULL`.

File: queries.sql [hunk 4]
```diff
@@ -17,4 +15,50 @@ select * from animals where neutered =true;
...
+-- Commit the transaction.
+COMMIT;
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (brytebee) [inline queries.sql:37]: - [x] **Required:** You are doing awesome. However, you must explicitly verify that the change was implemented before and after the rollback.

_**Hint**_ Consider `SELECT * FROM animals;`

File: queries.sql [hunk 5]
```diff
@@ -17,4 +15,50 @@ select * from animals where neutered =true;
...
+-- How many animals have never tried to escape?
+SELECT count(*) FROM animals WHERE escape_attempts = 0;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MAX(weight_kg), MIN(weight_kg) FROM animals GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-12-31' AND '2000-12-31' GROUP BY species;
```
Comment (brytebee) [inline queries.sql]: - [x] **Required:** The start date '1990-12-31' which you have specified selects from the last day in 1990, being the 31st of Dec, however, you are required to range from 1990. 
Kindly update it to 1st January 1990.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @joel-kalema👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
* Everything that creates or changes the structure of the tables is saved in the schema.sql file.✔️
* Everything that populates tables with data is saved in the data.sql file.✔️
* Everything that is a query that helps to analyze data is saved in the queries.sql file.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mre55 — APPROVED** [review]
**Hi @joel-kalema,**

# Status: APPROVED 🏆 🎉 🎊

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚀: 
Congratulations! 🎉 

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- The README is well documented ✅ 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**